analytics/src/analytics/integrations/github/validation.py

```diff
@@ -0,0 +1,119 @@
+"""Pydantic schemas for validating GitHub API responses."""
+
+# pylint: disable=no-self-argument
+from datetime import datetime, timedelta
+
+from pydantic import BaseModel, Field, computed_field, model_validator
+
+
+def safe_default_factory(data: dict, keys_to_replace: list[str]) -> dict:
+    """
+    Replace keys that are explicitly set to None with an empty dict for default_factory.
+
+    We need to do this because if a key is present, but its value is None or null,
+    it will raise a Pydantic ValidationError rather than using the default_factory.
+    """
+    for key in keys_to_replace:
+        if data.get(key) is None:
+            data[key] = {}
+    return data
```

Comment:
@coilysiren @chouinar I tried a number of ways to get around using something like this, but without spending another 1-2 hours researching this seems to be the best way to force a schema validator to use the `default_factory` when a key is present, but explicitly set to `None` (`null` in the JSON).

Reply:
This is reasonable to me 👍🏼